watertap/core/zero_order_base.py

```diff
@@ -0,0 +1,277 @@
+        'Process subtype to use when looking up parameters from database.'))
+
+    def build(self):
+        super().build()
+
+        # Set a placeholder attributes
+        # Placeholder for technology type string
+        self._tech_type = None
+
+        # Attribute indicating what parameters need to be fixed in the model
+        self._has_recovery_removal = False
+        self._fixed_perf_vars = []
+
+        # Place holders for assigning methods
+        self._initialize = None  # used to link to initization routine
+        self._scaling = None  # used to link to scaling routine
+        self._get_Q = None  # used to provide inlet volumetric flow
+
+        # Attributed for storing contents of reporting output
+        self._stream_table_dict = {}
+        self._perf_var_dict = {}
+
+        # Check that property package meets requirements
+        if self.config.property_package.phase_list != ["Liq"]:
+            raise ConfigurationError(
+                f"{self.name} configured with invalid property package. "
+                f"Zero-order models only support property packages with a "
+                f"single phase named 'Liq'.")
+        if (not hasattr(self.config.property_package, "solvent_set") or
+                self.config.property_package.solvent_set != ["H2O"]):
+            raise ConfigurationError(
+                f"{self.name} configured with invalid property package. "
+                f"Zero-order models only support property packages which "
+                f"include 'H2O' as the only Solvent.")
+        if not hasattr(self.config.property_package, "solute_set"):
+            raise ConfigurationError(
+                f"{self.name} configured with invalid property package. "
+                f"Zero-order models require property packages to declare all "
+                f"dissolved species as Solutes.")
+        if (len(self.config.property_package.solute_set) !=
+                len(self.config.property_package.component_list)-1):
+            raise ConfigurationError(
+                f"{self.name} configured with invalid property package. "
+                f"Zero-order models only support `H2O` as a solvent and all "
+                f"other species as Solutes.")
+
+    def initialize(self, state_args=None, outlvl=idaeslog.NOTSET,
+                   solver=None, optarg=None):
+        '''
+        Placeholder initialization routine, raises NotImplementedError
+        '''
+        if self._initialize is None or not callable(self._initialize):
+            raise NotImplementedError()
+        else:
+            self._initialize(self, state_args=None, outlvl=idaeslog.NOTSET,
+                             solver=None, optarg=None)
+
+    def calculate_scaling_factors(self):
+        '''
+        Placeholder scaling routine, should be overloaded by derived classes
+        '''
+        super().calculate_scaling_factors()
+
+        if callable(self._scaling):
+            self._scaling(self)
```

Comment:
Is this the unit specific calculate_scaling_factors()? I'd probably name it `_calculate_scaling_factors()`

Reply:
It is a private attribute that contains a reference to the unit specific `calculate_scaling_factors`. As a private attribute that most users will never see (let alone touch), I am not sure that naming really matters much.